#Issue34 Patch AI mode

TheRaj71 opened the pull request:
This pull request introduces a new game mode selection feature, including the ability to play against an AI opponent. It also includes several styling updates to support the new feature.

New game mode feature:

* [`paras/src/App.js`](diffhunk://#diff-4311fbab384bb893982d53521a3083128854d807ddf9f87e2e781b0786cf7defR14): Added state management for `gameMode` and implemented AI logic using the `minimax` algorithm to determine the best move for the AI player. [[1]](diffhunk://#diff-4311fbab384bb893982d53521a3083128854d807ddf9f87e2e781b0786cf7defR14) [[2]](diffhunk://#diff-4311fbab384bb893982d53521a3083128854d807ddf9f87e2e781b0786cf7defR76-R182)
* [`paras/src/App.js`](diffhunk://#diff-4311fbab384bb893982d53521a3083128854d807ddf9f87e2e781b0786cf7defL116): Updated the main component to conditionally render the game mode selection screen or the game board based on the selected game mode.

Styling updates:

* [`paras/src/App.css`](diffhunk://#diff-a1ccf801dd1fb56fa1e36afd47f997a0593423108441a6f630a4d48104681eedR291-R316): Added styles for the new game mode selection screen, including button styles and layout adjustments.

Code cleanup:


* [`paras/src/App.js`](diffhunk://#diff-4311fbab384bb893982d53521a3083128854d807ddf9f87e2e781b0786cf7defL26-R29): Minor formatting changes for readability in the `checkWinner` function.

Here's a sample video of new feature attached. @iamparas0 Is this good ??

https://github.com/user-attachments/assets/5a9163ab-af23-46aa-a9af-d577dda3a5c7

iamparas0 commented:
> This pull request introduces a new game mode selection feature, including the ability to play against an AI opponent. It also includes several styling updates to support the new feature.
> 
> New game mode feature:
> 
> * [`paras/src/App.js`](https://github.com/iamparas0/TIC-TAC-TOE/pull/101/files#diff-4311fbab384bb893982d53521a3083128854d807ddf9f87e2e781b0786cf7defR14): Added state management for `gameMode` and implemented AI logic using the `minimax` algorithm to determine the best move for the AI player. [[1]](https://github.com/iamparas0/TIC-TAC-TOE/pull/101/files#diff-4311fbab384bb893982d53521a3083128854d807ddf9f87e2e781b0786cf7defR14) [[2]](https://github.com/iamparas0/TIC-TAC-TOE/pull/101/files#diff-4311fbab384bb893982d53521a3083128854d807ddf9f87e2e781b0786cf7defR76-R182)
> * [`paras/src/App.js`](https://github.com/iamparas0/TIC-TAC-TOE/pull/101/files#diff-4311fbab384bb893982d53521a3083128854d807ddf9f87e2e781b0786cf7defL116): Updated the main component to conditionally render the game mode selection screen or the game board based on the selected game mode.
> 
> Styling updates:
> 
> * [`paras/src/App.css`](https://github.com/iamparas0/TIC-TAC-TOE/pull/101/files#diff-a1ccf801dd1fb56fa1e36afd47f997a0593423108441a6f630a4d48104681eedR291-R316): Added styles for the new game mode selection screen, including button styles and layout adjustments.
> 
> Code cleanup:
> 
> * [`paras/src/App.js`](https://github.com/iamparas0/TIC-TAC-TOE/pull/101/files#diff-4311fbab384bb893982d53521a3083128854d807ddf9f87e2e781b0786cf7defL26-R29): Minor formatting changes for readability in the `checkWinner` function.
> 
> Here's a sample video of new feature attached. @iamparas0 Is this good ??
> 
>  tictactoe.mp4

you did great jo , kindly requesting you to create back button so if we get bored so we can choose other options

TheRaj71 commented:
> > This pull request introduces a new game mode selection feature, including the ability to play against an AI opponent. It also includes several styling updates to support the new feature.
> > New game mode feature:
> > 
> > * [`paras/src/App.js`](https://github.com/iamparas0/TIC-TAC-TOE/pull/101/files#diff-4311fbab384bb893982d53521a3083128854d807ddf9f87e2e781b0786cf7defR14): Added state management for `gameMode` and implemented AI logic using the `minimax` algorithm to determine the best move for the AI player. [[1]](https://github.com/iamparas0/TIC-TAC-TOE/pull/101/files#diff-4311fbab384bb893982d53521a3083128854d807ddf9f87e2e781b0786cf7defR14) [[2]](https://github.com/iamparas0/TIC-TAC-TOE/pull/101/files#diff-4311fbab384bb893982d53521a3083128854d807ddf9f87e2e781b0786cf7defR76-R182)
> > * [`paras/src/App.js`](https://github.com/iamparas0/TIC-TAC-TOE/pull/101/files#diff-4311fbab384bb893982d53521a3083128854d807ddf9f87e2e781b0786cf7defL116): Updated the main component to conditionally render the game mode selection screen or the game board based on the selected game mode.
> > 
> > Styling updates:
> > 
> > * [`paras/src/App.css`](https://github.com/iamparas0/TIC-TAC-TOE/pull/101/files#diff-a1ccf801dd1fb56fa1e36afd47f997a0593423108441a6f630a4d48104681eedR291-R316): Added styles for the new game mode selection screen, including button styles and layout adjustments.
> > 
> > Code cleanup:
> > 
> > * [`paras/src/App.js`](https://github.com/iamparas0/TIC-TAC-TOE/pull/101/files#diff-4311fbab384bb893982d53521a3083128854d807ddf9f87e2e781b0786cf7defL26-R29): Minor formatting changes for readability in the `checkWinner` function.
> > 
> > Here's a sample video of new feature attached. @iamparas0 Is this good ??
> > tictactoe.mp4
> 
> you did great job , kindly requesting you to create back button so if we get bored so we can choose other options

Added a simple back button redirecting user to home as per feedback 😊. Below is attachment
![Screenshot 2024-10-06 233821](https://github.com/user-attachments/assets/f0f9536d-783c-45de-8735-b0b4d520bd30)

iamparas0 commented:
> > > This pull request introduces a new game mode selection feature, including the ability to play against an AI opponent. It also includes several styling updates to support the new feature.
> > > New game mode feature:
> > > 
> > > * [`paras/src/App.js`](https://github.com/iamparas0/TIC-TAC-TOE/pull/101/files#diff-4311fbab384bb893982d53521a3083128854d807ddf9f87e2e781b0786cf7defR14): Added state management for `gameMode` and implemented AI logic using the `minimax` algorithm to determine the best move for the AI player. [[1]](https://github.com/iamparas0/TIC-TAC-TOE/pull/101/files#diff-4311fbab384bb893982d53521a3083128854d807ddf9f87e2e781b0786cf7defR14) [[2]](https://github.com/iamparas0/TIC-TAC-TOE/pull/101/files#diff-4311fbab384bb893982d53521a3083128854d807ddf9f87e2e781b0786cf7defR76-R182)
> > > * [`paras/src/App.js`](https://github.com/iamparas0/TIC-TAC-TOE/pull/101/files#diff-4311fbab384bb893982d53521a3083128854d807ddf9f87e2e781b0786cf7defL116): Updated the main component to conditionally render the game mode selection screen or the game board based on the selected game mode.
> > > 
> > > Styling updates:
> > > 
> > > * [`paras/src/App.css`](https://github.com/iamparas0/TIC-TAC-TOE/pull/101/files#diff-a1ccf801dd1fb56fa1e36afd47f997a0593423108441a6f630a4d48104681eedR291-R316): Added styles for the new game mode selection screen, including button styles and layout adjustments.
> > > 
> > > Code cleanup:
> > > 
> > > * [`paras/src/App.js`](https://github.com/iamparas0/TIC-TAC-TOE/pull/101/files#diff-4311fbab384bb893982d53521a3083128854d807ddf9f87e2e781b0786cf7defL26-R29): Minor formatting changes for readability in the `checkWinner` function.
> > > 
> > > Here's a sample video of new feature attached. @iamparas0 Is this good ??
> > > tictactoe.mp4
> > 
> > 
> > you did great job , kindly requesting you to create back button so if we get bored so we can choose other options
> 
> Added a simple back button redirecting user to home as per feedback 😊. Below is attachment is it good to go now @iamparas0 ![Screenshot 2024-10-06 233821](https://private-user-images.githubusercontent.com/143867566/373969753-f0f9536d-783c-45de-8735-b0b4d520bd30.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgyODcyNzEsIm5iZiI6MTcyODI4Njk3MSwicGF0aCI6Ii8xNDM4Njc1NjYvMzczOTY5NzUzLWYwZjk1MzZkLTc4M2MtNDVkZS04NzM1LWIwYjRkNTIwYmQzMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwN1QwNzQyNTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xOTA5MzEyNzRkZTBjOTAyNjM2MDY4ODU5OWE1Zjk3MTk2MDhkZTg0NDE2ODVmNjNjOWY0NWIxNTE3YmQwMzM4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.8EESTP4nRuIHivOH5ThXFdmNsyxsjcY2q6I5IMHAeCk)

Please create pr again so I can merge it
